### PR TITLE
refactor: Fix pre-commit errors when adding new python versions

### PR DESCRIPTION
**Pull request by david-k-johnson**
Fixed all pre-commit errors that pre-existed
Add python 3.10
Added Pre-commit to github actions
Added conf files for developers to use
Added missing types

### CODE REVIEW

**Comment by david-k-johnson** [comment]
It would be great if I could get my PR to run the workflows without approval:). Just trying to fix the state of the world and improve on the CI

**Comment by david-k-johnson** [comment]
I had to roll back 3.11 which seemed to pass tests locally for me, but since dbt does not officially support 3.11 and it failed in our github action I just reverted.  Also it looks like the workflows seem to be mostly passing in our fork except the Main CI task fails when it gets to reading the secret we do not have setup.  Anyway I know this is a bigger change, but it kind of snowballed as I was adding 3.10 support I found lots of other issues since pre-commit was failing for me.  I fixed all the pre-commit issues and I think it cleaned up the code a bit and fixed some questionable code as well.  Hope this PR is helpful.

**Comment by david-k-johnson** [comment]
I have seen no activity or questions on this, since this is a large change that fixes all the pre-commit issues, this will have to be merged quickly or it will rot.  I think this fixes many issues with the repo(pre-commit, ci, actual bugs) all found by flake8 and mypy.  Again I leave you to the owners to decide

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[flake8]
+max-line-length = 120
+ignore = ['D100','D101','D102','D103','D104','D401','D400','D107','W503','D200']
```
Comment (ssassi) [inline .flake8:3]: Why did you need to ignore all these rules?

Comment (david-k-johnson) [reply .flake8:3]: I did not add that many, if you look in .pre-commit-config.yaml you were already skipping D102,D104,D401,D400,D107,W503, so I just added the following to the flake8 config so that users running in their editor get the same rules applied as pre-commit is using.

D100:Missing docstring in public module
D101:Missing docstring in public class
D103:Missing docstring in public function
D200:One-line docstring should fit on one line with quotes

All of these are doc string related and I do not feel qualifies to add all the missing doc strings.  In addition I already fixed hundreds of errors since the repo is clearly never running pre-commit checks.  I would suggest merging my chnages and making a seperate PR to clear up all skipped errors.

**Comment by david-k-johnson** [comment]
So I guess I should just close this?  I see someone(@ssassi) created an issue to check pre-commit in ci which I had already did and fixed all issues in this PR, but now it has rotted(BTW that was what the majority of this commit was, fixing pre-commit issues) If I get no response soon I will just close this PR.

**Comment by david-k-johnson** [comment]
BTW my PR also fixed all mypy issues, but I guess you did want me modifying every file:)

**Comment by ssassi** [comment]
Thank you @david-k-johnson for your PR. We noticed your branch diverged from main and we ended up doing it ourselves, partially. We should've merged yours earlier as you suggested 😬.
